scam-detector-py/src/agent.py

```diff
@@ -936,6 +938,110 @@ def emit_new_fp_finding(w3) -> list:
+                page_size = math.floor(page_size / 2)
+                should_retry_from_error = page_size > 1
+            else:               
+                logging.warning(f"{BOT_VERSION}: update reactive likely fps (get_alerts error): {e} - {traceback.format_exc()}")
+                Utils.ERROR_CACHE.add(Utils.alert_error(str(e), "agent.update_reactive_likely_fps.internal1", traceback.format_exc()))
+        if (not should_retry_from_error and response.page_info.end_cursor.alert_id == ""):
+            break
+        
+    logging.info(f"{BOT_VERSION}: update reactive likely fps (alerts count): {len(alerts)}")
+    
+    unique_scammers_list = Utils.process_past_alerts(alerts, REACTIVE_LIKELY_FPS, BOT_VERSION)
+    labels = []
+    page_size = 1000
+    batch_size = 200
+    for i in range(0, len(unique_scammers_list), batch_size):
+        batch = unique_scammers_list[i:i+batch_size]
+        
+        labels_response = None
+        starting_cursor = None
+        should_retry_from_error = False
+
+        while True:
+            if labels_response and labels_response.page_info and labels_response.page_info.has_next_page:
+                starting_cursor = labels_response.page_info.end_cursor      
+                should_retry_from_error = False
+            try:
+                query = {
+                    "entities": batch,
+                    "source_ids": [source_id],
+                    "labels": ["scammer"],
+                    "created_since": start_milliseconds_ago,
+                    "first": page_size,
+                    "starting_cursor": starting_cursor
+                }
+
+                labels_response = get_labels(query)
+                labels.extend(labels_response.labels)
+            except Exception as e:
+                if (
+                    (isinstance(e, AttributeError) and 'NoneType' in str(e)) or
+                    (isinstance(e, Exception) and "Internal server error" in str(e))
+                ):
+                    # Reduce the page size in order to reduce the response size and try again
+                    page_size = math.floor(page_size / 2)
+                    should_retry_from_error = page_size > 1
+                else:               
+                    logging.warning(f"{BOT_VERSION}: update reactive likely fps (get_labels error): {e} - {traceback.format_exc()}")
+                    Utils.ERROR_CACHE.add(Utils.alert_error(str(e), "agent.update_reactive_likely_fps.internal3", traceback.format_exc()))
+                    raise e
+                
+            if not should_retry_from_error and not labels_response.page_info.has_next_page:
+                break
+
+    logging.info(f"{BOT_VERSION}: update reactive likely fps (labels count): {len(labels)}")
+
+    for label in labels:
+        if not label.remove and label.entity not in REACTIVE_LIKELY_FPS:
+            REACTIVE_LIKELY_FPS[label.entity] = label.metadata
```

Comment:
so this dict will contain all scammer addresses that were added 89 days ago for a given hour

in handle block, you then check whether it is an FP, right?

Reply:
That's correct. (`REACTIVE_LIKELY_FPS` would also contain any possible unprocessed labels from the previous hour period(s))

Reply:
why not check for FP before you put it in the REACTIVE_LIKELY_FPS dictionary?